nocodb/infra/requests_client.py

```diff
@@ -12,13 +13,33 @@
 
 
 class NocoDBRequestsClient(NocoDBClient):
-    def __init__(self, auth_token: AuthToken, base_uri: str):
+    def __init__(
+        self,
+        base_uri: str,
+        auth_token: AuthToken = None,
+        email: str = None,
+        password: str = None,
+    ):
         self.__session = requests.Session()
+        self.__api_info = NocoDBAPI(base_uri)
+
+        if not auth_token and not (email and password):
+            raise ValueError("Either сredentials or token must be provided")
+
+        if not auth_token and (email and password):
+            auth_token = JWTAuthToken(self.get_auth_token(email, password))
```

Comment:
I'm not sure about having a HTTP request in the `NocoDBRequestsClient` initialization. I suggest creating a class with the purpose of retrieving the AuthToken.

```mermaid
classDiagram
class NocoDBAuthClient {
  << interface >>
   +get_auth_token() AuthToken
}


class NocoDBAuthRequestsClient {
   +get_auth_token() AuthToken
}

NocoDBAuthClient <|-- NocoDBAuthRequestsClient

```

Usage example:

```python
auth_client = NocoDBAuthRequestsClient("username", "password")
auth_token = auth_client.get_auth_token()
client = NocoDBRequestsClient(base_uri, auth_token)
```